crates/biome_js_analyze/src/lint/nursery/use_explicit_type.rs

```diff
@@ -202,22 +228,87 @@ declare_lint_rule! {
     /// }
     /// ```
     ///
+    /// The following pattern is considered correct code because it includes both a return type and an argument type on an function:
+    ///
+    /// ```ts
+    /// export function test(a: number, b: number): void {
+    ///   return;
+    /// }
+    /// ```
+    ///
+    /// ```ts
+    /// export default function test(obj: {a: string}): void {
+    ///   return;
+    /// }
+    /// ```
+    ///
+    /// ```ts
+    /// export default (a: SomeType): void => {
+    ///   return;
+    /// }
+    /// ```
+    ///
     pub UseExplicitType {
         version: "1.9.3",
         name: "useExplicitType",
         language: "ts",
         recommended: false,
-        sources: &[RuleSource::EslintTypeScript("explicit-function-return-type")],
+        sources: &[RuleSource::EslintTypeScript("explicit-function-return-type"), RuleSource::EslintTypeScript("explicit-module-boundary-types")],
     }
 }
 
 declare_node_union! {
-    pub AnyJsFunctionWithReturnType = AnyJsFunction | JsMethodClassMember | JsMethodObjectMember | JsGetterClassMember | JsGetterObjectMember
+    pub AnyJsFunctionWithReturnTypeOrJsParameters = AnyJsFunction | JsMethodClassMember | JsMethodObjectMember | JsGetterClassMember | JsGetterObjectMember | JsParameters
+}
+
+pub struct UseExplicitTypeState {
+    range: TextRange,
+    cause: UseExplicitTypeCause,
+}
+
+enum UseExplicitTypeCause {
+    MissingReturnType,
+    MissingArgumentnType(String),
```

Comment:
Usually we try to avoid allocating a string. You could directly take the token:

```suggestion
    MissingArgumentnType(JsSyntaxToken),
```

In this case we could rewrite `UseExplicitTypeState` as:

```rs
pub enum UseExplicitTypeState {
    MissingReturnType(TextRange),
    MissingArgumentnType(JsSyntaxToken),
}
```

Because we can get the range from a token.